### PR TITLE
Added notification cooldown setting in YAML and logic to use it.

### PR DESCRIPTION
**Pull request by matacoder**
Resolves: #28

Using `notifier_frequency: 720` in `blackbox.yaml` we set how many notifications we want to see inside one day.

Logic is simple - keep last time of notification and check if we can send. Otherwise increase integer value of attempts to show it later in logs.

First start, sending:
```
blackbox    | 06:56:11 Recreating json with period: 28800
```

Frequency in YAML changed, sending:
```
blackbox    | 07:05:00 Found json, reading it...
blackbox    | 07:05:00 Frequency changed. Recreating json with period: 120
```

Unsuccessful attempt to send:
```
blackbox    | 07:06:51 Found json, reading it...
blackbox    | 07:06:51 Can't send notification, 111 < 120.
```

Successful attempt to send after period:
```
blackbox    | 07:07:07 Found json, reading it...
blackbox    | 07:07:07 Recreating json with period: 120
blackbox    | 07:07:07 By the way there were 6 successful backups.
```

I need confirmation that logic is okay and location of JSON is appropriate and then I will update README file.

### CODE REVIEW

**Comment by lemonsaurus** [comment]
I don't think I like this approach, it just feels wrong to me to specify it as a number of notifications per day. I don't want to force the users to do maths in order to figure out how to configure this. It just seems like bad user experience.

I envisioned this feature more like a _cooldown_ for success notifications that the user would configure. The user sets the cooldown to one hour, and every subsequent success notification will happen no sooner than 1 hour after the previous one.

I think it's more intuitive to think of this in terms of time, rather than number of notifications in a day.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
I'd prefer a cooldown-based approach.

**Comment by matacoder** [comment]
I've changed logic to cooldown as you asked.
## Cooldown

By default, `blackbox` will send all notification at every backup attempt. 
You can specify a `cooldown` period in `blackbox.yaml` during which all notifications will be muted.
This option will not mute failed backups.

### Example usage

```yaml
cooldown: 120s
```
```yaml
cooldown: 3 hours
```
```yaml
cooldown: 2 days 4 hours
```
```yaml
cooldown: 4h 32M 16s
```

File: tests/test_cooldown.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import pytest
+from dateutil.relativedelta import relativedelta
+
+from blackbox.utils.cooldown import parse_config_cooldown
+from blackbox.utils.cooldown import should_we_send
+
+
+@pytest.fixture
+def cooldown_config():
+    """Test human view cool down period."""
```
Comment (lemonsaurus) [inline tests/test_cooldown.py:12]:
```suggestion
    """Test humanized cooldown period."""
```

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Minor nitpicking left to fix, and then this is good. 

I really like this overall approach now, well done! 👏🏼 👏🏼👏🏼

**Review by lemonsaurus — APPROVED** [review]
Looks good!

**Pull request merged**